### PR TITLE
feat(Dialog): migrate to tailwind

### PR DESCRIPTION
**Pull request by mainframev**
Migrate Drawer to Tailwind 

[FEPLT-1661](https://kiwicom.atlassian.net/browse/FEPLT-1661)
 Storybook: https://orbit-mainframev-tw-migrate-dialog.surge.sh

### CODE REVIEW

File: packages/orbit-components/src/Dialog/index.tsx [hunk 1]
```diff
@@ -150,51 +52,86 @@ const Dialog = ({
...
+        aria-modal="true"
+        aria-labelledby={dialogID}
+        aria-describedby={descriptionId}
+        className="flex min-h-full items-center"
+      >
+        <div
+          ref={ref}
+          id={dialogID}
+          style={{ maxWidth }}
+          className={cx([
```
Comment (DSil) [inline packages/orbit-components/src/Dialog/index.tsx:93]: Why make it array?

Comment (DSil) [reply packages/orbit-components/src/Dialog/index.tsx:93]: This is missing some properties:
```
text-align: center;
.orbit-text {
  text-align: center;
}

largeMobile:
text-align: start;
.orbit-text {
  text-align: start;
}
```

**EDIT:** I see this responsability was added to the Stack used inside. ~I am not sure about how we can ensure there are no side-effects on this~ I am not sure we can do it, because one applies to items and others to text. And we can have more than just text inside the Dialog.

File: packages/orbit-components/src/Dialog/index.tsx [hunk 2]
```diff
@@ -150,51 +52,86 @@ const Dialog = ({
...
+            "shadow-overlay pt-lg px-md pb-md bg-white-normal box-border block w-full rounded-[12px]",
+            shown ? "bottom-0" : "-bottom-full",
+            "lm:min-w-[540px] lm:p-lg lm:rounded-[9px]",
+          ])}
+        >
+          {illustration && <div className="mb-md">{illustration}</div>}
+          <Stack
+            spacing="XSmall"
+            direction="column"
```
Comment (DSil) [inline packages/orbit-components/src/Dialog/index.tsx:102]: Why was this added?

File: packages/orbit-components/src/Dialog/index.tsx [hunk 3]
```diff
@@ -150,51 +52,86 @@ const Dialog = ({
...
           </Stack>
           <Stack
             direction="column-reverse"
             spacing="XSmall"
             largeMobile={{ direction: "row", justify: "end" }}
           >
-            {secondaryAction && <StyledAction>{secondaryAction}</StyledAction>}
-            <StyledAction>{primaryAction}</StyledAction>
+            {secondaryAction && (
+              <div className="lm:w-auto lm:[&>button]:flex-none lm:[&>button]:w-auto [button]:w-full w-full [&>button]:flex-auto">
```
Comment (DSil) [inline packages/orbit-components/src/Dialog/index.tsx:124]: I think it is
```suggestion
              <div className="lm:w-auto lm:[&>button]:flex-none lm:[&>button]:w-auto [&>button]:w-full w-full [&>button]:flex-auto">
```
no?

Also, this is reused, why is it not abstracted?

File: packages/orbit-components/src/Dialog/index.tsx [hunk 4]
```diff
@@ -150,51 +52,86 @@ const Dialog = ({
...
+                {secondaryAction}
+              </div>
+            )}
+            <div className="lm:w-auto lm:[&>button]:flex-none lm:[&>button]:w-auto w-full [&>button]:w-full [&>button]:flex-auto">
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx:128]: @DSil maybe, but this should not be critical, as I doubt it will change anything even if we will keep normal selector

File: packages/orbit-components/src/Dialog/index.tsx (hunk 2 above)
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx:102]: In order to avoid alignment with css selectors, as it was before

File: packages/orbit-components/src/Dialog/index.tsx [hunk 5]
```diff
@@ -150,51 +52,86 @@ const Dialog = ({
...
       tabIndex={0}
-      role="dialog"
-      aria-modal="true"
-      aria-labelledby={dialogID}
+      role="button"
+      className={cx([
+        "font-base",
+        "h-full w-full",
+        "fixed bottom-0 left-0 right-0 top-0",
+        "p-md z-overlay box-border overflow-x-hidden bg-[rgba(0,0,0,0.5)]",
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx:77]: it is, but I decided to have it like that, probably should added a comment to make it clear in advance.
The thing is, that those bgs are the same for Dialog, Modal, Drawer and othey components that have overlay. So I would unify that logic when we will refactor Modal component, having a single overlay token would be better, than having 3 for the same thing as I doubt that we will need to have different overlays for those components.

File: packages/orbit-components/src/Dialog/index.tsx [hunk 6]
```diff
@@ -150,51 +52,86 @@ const Dialog = ({
...
     >
-      <StyledDialogCenterWrapper>
-        <StyledDialogContent shown={shown} ref={ref} id={dialogID} $maxWidth={maxWidth}>
-          {illustration && <IllustrationContainer>{illustration}</IllustrationContainer>}
-          <Stack spacing="XSmall" spaceAfter="medium">
+      <div
+        role="dialog"
+        aria-modal="true"
+        aria-labelledby={dialogID}
+        aria-describedby={descriptionId}
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx:86]: see the comment above

File: packages/orbit-components/src/Dialog/index.tsx [hunk 7]
```diff
@@ -150,51 +52,86 @@ const Dialog = ({
...
+        className="flex min-h-full items-center"
+      >
+        <div
+          ref={ref}
+          id={dialogID}
+          style={{ maxWidth }}
+          className={cx([
+            "shadow-overlay pt-lg px-md pb-md bg-white-normal box-border block w-full rounded-[12px]",
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx]: It was quicker for me to do that without tokens, as still I think not necceserily we should everything put into tokens, but still added here as you suggested :)

File: packages/orbit-components/src/Dialog/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-components/src/Dialog/index.tsx]: Right, so we can create that global token in this PR and start using it here, no? 🤔

File: packages/orbit-components/src/Dialog/index.tsx [hunk 8]
```diff
@@ -150,51 +52,86 @@ const Dialog = ({
...
       ref={wrapperRef}
       data-test={dataTest}
+      onKeyDown={handleKeyDown}
       id={id}
       onClick={handleClose}
       tabIndex={0}
-      role="dialog"
-      aria-modal="true"
-      aria-labelledby={dialogID}
+      role="button"
```
[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-components/src/Dialog/index.tsx:80]: I'm sorry but I couldn't find a reference for that, can you please point me to that? 🙏  It is not clear to me why the whole dialog wrapper should have `role="button"`

I am in favor of fixing accessibility issues we might have had in the past, but this one is not clear to me

File: packages/orbit-components/src/Dialog/index.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-components/src/Dialog/index.tsx]: Yes, I understand, I wouldn't be blocking for this. Honestly, I think it depends on the property. For instance, in other components I did not mind having a custom height directly there, with no token. But in this case, as we also have for badge, I think a token is more explicit about what we want to achieve. Thanks 👍

File: packages/orbit-components/src/Dialog/index.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx:80]: > It is not clear to me why the whole dialog wrapper should have role="button"

Because the whole wrapper is clickable (it's just overlay) and it has to have appropriate role and onClick and onKeyDown, it's ok to have it as button. If the main wrapper will have role dialog (it's non interactive role) it will have a warning from [eslint a11y rule
](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/3d1d26d89d492947cbf69f439deec9e7cfaf9867/docs/rules/no-noninteractive-element-interactions.md) 

For dialog itself is important to have:
- container that serves as dialog main wrapper should have role="dialog"
- some element inside should be focused after it's opened (autofocus)
- aria-label and aria-describedby 

Now it's missing the autofocus on opened, I think we can autofocus the primary action, wdyt? 

In general, I think if the keyboard navigation is ok and screen reader reads correctly everything, then everything should be 👌🏻

[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx]: Yep, I found it [here](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/) and it says that the main content of dialog should have role dialog. In the previous implementation it was throwing a warning, as Dialog can't be interactive element (having onClick and etc), but you also have a fair point, so I slightly adjusted that to have a common wrapper with only role and aria attributes.

File: packages/orbit-components/src/Dialog/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx]: I would prefer to do that in the separate PR, if you do not mind :)

File: packages/orbit-components/src/Dialog/index.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx]: The another thing I am thinking about is that probably we should autofocus the primary action, when it's opening.

File: packages/orbit-components/src/Dialog/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-components/src/Dialog/index.tsx]: I don't. But then why is the `dialogOverlayBackground` token being created here? And that means keeping the custom value for now, right? I am ok, just worried to have dangling custom values around

[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx]: You are right, dropped them ;)

File: packages/orbit-components/src/Dialog/index.tsx [hunk 9]
```diff
@@ -150,51 +67,74 @@ const Dialog = ({
...
+        data-test={dataTest}
+        onKeyDown={handleKeyDown}
+        id={id}
+        onClick={handleClose}
+        tabIndex={0}
+        role="button"
+        className={cx([
+          "font-base",
+          "h-full w-full",
+          "p-md z-overlay bg-dialog-overlay-background box-border overflow-x-hidden",
```
Comment (DSil) [inline packages/orbit-components/src/Dialog/index.tsx:97]: This bg now does not exist :sweat: You ended up changing but also removing the token?

File: packages/orbit-components/src/Dialog/index.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-components/src/Dialog/index.tsx]: My concern is about the `role="button"` on the whole Dialog, which I think is not correct. It is saying the whole dialog + overlay is clickable, which is not true. It also applies (by preflight rules) a cursor pointer to the whole dialog and overlay, which are also undersired

File: packages/orbit-components/src/Dialog/index.tsx (hunk 9 above)
Comment (mainframev) [reply packages/orbit-components/src/Dialog/index.tsx]: I removed the token, yes, now changed to the normal bg-static value 👍🏻 Thanks
